Game.java

```diff
@@ -0,0 +1,250 @@
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.Vector;
+
+public class Game{
+	
+	private Board topBoard, botBoard;
+	
+	boolean gameOn;
+	
+	public static final Direction defaultDirection = Direction.Left;
+	
+	Timer moveTimer;
+	
+	public Game(){
+		this(null); // prevents duplicate constructors and mistakes
+	}
+	
+	public Game(final String[] b){
+		this.gameOn = false;
+		this.moveTimer = new Timer();
+				
+		this.topBoard = new Board(b);
+		this.topBoard.limitToSpecificCreatures(new Board.Creature[]{
+				Board.Creature.Ghost1, Board.Creature.Ghost2, Board.Creature.Ghost3,
+				Board.Creature.Ghost4, Board.Creature.Pacman});
+		
+		this.botBoard = new Board(b);
+		this.botBoard.limitToSpecificCreatures(new Board.Creature[]{
+				Board.Creature.Wall, Board.Creature.Point, Board.Creature.Null} );
+	
+	}	
+	
+	enum Direction{
+		Right,
+		Left,
+		Up,
+		Down
+	}
+	
+	private void movePacman(){	
+		Board.Creature p = Board.Creature.Pacman;
+		Location currentLocation = p.getLocation();
+		Location nextLocation = getNextLocation(p, p.getDirection());
+		
+		switch ( getCreatureAt(nextLocation) ){
+		case Point:
+			// score ++
+		case Null:
+			topBoard.set(currentLocation, Board.Creature.Null);
+			topBoard.set(nextLocation, p);
+			botBoard.set(nextLocation, Board.Creature.Null);
+			break;
+			
+		case Ghost1:
+		case Ghost2:
+		case Ghost3:
+		case Ghost4:
+			stopGame( p );
+			break;
+			
+		default:
+			break;
+		}
+	}
+
+	private Location getNextLocation(Board.Creature c, Direction d) {
+		Location nextLocation = changeLocationByDirection(c.getLocation().getX(), c.getLocation().getY(), d);
+		
+		int Xdim = topBoard.getDimensions(), Ydim = Xdim;
+		
+		if (topBoard.get().length != topBoard.get()[0].length){
```

Comment:
Why using the board array itself instead of asking for the data that you're interesting of (the length)?